disnake/channel.py

```diff
@@ -1817,6 +1853,479 @@ class NewsChannel(TextChannel):
+        The channel's ID.
+    name: :class:`str`
+        The channel's name.
+    guild: :class:`Guild`
+        The guild the channel belongs to.
+    topic: Optional[:class:`str`]
+        The channel's topic. ``None`` if it isn't set.
+    category_id: Optional[:class:`int`]
+        The category channel ID this channel belongs to, if applicable.
+    position: :class:`int`
+        The position in the channel list. This is a number that starts at 0. e.g. the
+        top channel is position 0.
+    nsfw: :class:`bool`
+        Whether the channel is marked as "not safe for work".
+
+        .. note::
+
+            To check if the channel or the guild of that channel are marked as NSFW, consider :meth:`is_nsfw` instead.
+    last_thread_id: Optional[:class:`int`]
+        The ID of the last created thread in this channel. It may
+        *not* point to an existing or valid thread.
+    default_auto_archive_duration: :class:`int`
+        The default auto archive duration in minutes for threads created in this channel.
+    """
+
+    # TODO: Refer threads as posts?
+
+    __slots__ = (
+        "id",
+        "name",
+        "category_id",
+        "topic",
+        "position",
+        "nsfw",
+        "last_thread_id",
+        "default_auto_archive_duration",
+        "guild",
+        "_state",
+        "_type",
+        "_overwrites",
+    )
+
+    def __init__(self, *, state: ConnectionState, guild: Guild, data: ForumChannelPayload) -> None:
+        self._state: ConnectionState = state
+        self.id: int = int(data["id"])
+        self._type: int = data["type"]
+        self._update(guild, data)
+
+    def __repr__(self) -> str:
+        atts = [
+            ("id", self.id),
+            ("name", self.name),
+            ("topic", self.topic),
+            ("position", self.position),
+            ("nsfw", self.nsfw),
+            ("category_id", self.category_id),
+            ("default_auto_archive_duration", self.default_auto_archive_duration),
+        ]
+        joined = " ".join("%s=%r" % t for t in atts)
+        return f"<{type(self).__name__} {joined}>"
+
+    def _update(self, guild: Guild, data: ForumChannelPayload) -> None:
+        self.guild: Guild = guild
+        self.name: str = data["name"]
+        self.category_id: Optional[int] = utils._get_as_snowflake(data, "parent_id")
+        self.topic: Optional[str] = data.get("topic")
+        self.position: int = data["position"]
+        self.nsfw: bool = data.get("nsfw", False)
+        self.last_thread_id: Optional[int] = utils._get_as_snowflake(data, "last_message_id")
+        self.default_auto_archive_duration: ThreadArchiveDurationLiteral = data.get(
+            "default_auto_archive_duration", 1440
+        )
+        self._fill_overwrites(data)
+
+    async def _get_channel(self) -> ForumChannel:  # I wonder why this exists
+        return self
+
+    typing = disnake.abc.Messageable.typing
+    trigger_typing = disnake.abc.Messageable.trigger_typing
```

Comment:
While these are supported, they should have better docstrings than what they currently do by just being reassigned. I'll update this.

Reply:
This was changed in b63f5836cb8cfbac17c2f67b9cae391439572b50, thanks arl